apps/exam/views.py

```diff
@@ -50,16 +50,21 @@ def get_context_data(self, **kwargs):
     context['exam'] = exam
     context[
         'attempt_duration'] = exam.duration_minutes + self.request.user.required_extra_time
+    context['exam_mode'] = int(self.kwargs['exam_mode'])
```

Comment:
The `exam` already contains  the `mode` information. There is no need to add it to the context explicitly.